### PR TITLE
Added Payment Method class

### LINKED ISSUE #3: PaymentMethod
The PaymethMethod class represents a payment method used for a transaction, it should contain an attribute (most likely a String) that indicates the methode of payment used.
As specified in the class diagram, a Transaction object will contain a PaymentMethod object.

### PR DESCRIPTION
**Pull request by Banalian**
closes #3

I took the liberty of creating specific isMethod for the current known methods, but maybe they should be removed since there is the general one ?

### CODE REVIEW

File: src/main/java/fr/ppptours/cafeteria/model/paymentmethod/Method.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * Represents a payment method.
+ */
+public enum Method {
+    CASH,
+    CARD,
+    LYDIA,
+    ACCOUNT
+}
```
Comment (Nowp) [inline src/main/java/fr/ppptours/cafeteria/model/paymentmethod/Method.java:11]: Not good, this is not very evolutive

**Review by Nowp — CHANGES_REQUESTED** [review]
Why create a class if our values are only defined by the Method enum ?
Payment method should only have a name at least.

File: src/main/java/fr/ppptours/cafeteria/model/paymentmethod/Method.java (hunk 1 above)
Comment (Banalian) [reply src/main/java/fr/ppptours/cafeteria/model/paymentmethod/Method.java:11]: So we'd just have a PaymentMethod class with a name variable, and the check/get function ?

Comment (Nowp) [reply src/main/java/fr/ppptours/cafeteria/model/paymentmethod/Method.java:11]: Just a class with a name lilianx

**Review by Nowp — CHANGES_REQUESTED** [review]

**Review by jumsee — APPROVED** [review]
Seems good for me

**Review by AdamWizard — APPROVED** [review]
Looks okay to me.

File: src/main/java/fr/ppptours/cafeteria/model/paymentmethod/PaymentMethod.java [hunk 2]
```diff
@@ -6,115 +6,46 @@
...
-    private Method method;
-
-    /**
-     * Creates a payment method from a given method
-     * @param method The method to use
-     */
-    public PaymentMethod(Method method) {
-        this.method = method;
-    }
+    private String method;
```
Comment (Nowp) [inline src/main/java/fr/ppptours/cafeteria/model/paymentmethod/PaymentMethod.java:11]: rename to name

File: src/main/java/fr/ppptours/cafeteria/model/paymentmethod/PaymentMethod.java [hunk 3]
```diff
@@ -6,115 +6,46 @@
...
+     * Checks if the payment method given is the same as the object's one
      * @param method the payment method to compare to
      * @return true if the payment method is the same, false otherwise
-     * @throws IllegalArgumentException if the given string is not a valid payment method
      */
     public boolean isMethod(String method) {
-        return this.method == parseMethodFromString(method);
+        String methodLower = method.toLowerCase();
+        return this.method.toLowerCase().equals(methodLower);
     }
```
[the remark this answers is not in the document]
Comment (Banalian) [reply src/main/java/fr/ppptours/cafeteria/model/paymentmethod/PaymentMethod.java]: would id also be a String or a Int ?

**Review by AdamWizard — APPROVED** [review]

**Review by Nowp — APPROVED** [review]

**Review by jumsee — COMMENTED** [review]
Seems good

**Pull request merged**